File: docs/examples/java/output/parts/UserManagementService_validateEmail.java
// code2map fragment (non-buildable)
// original: docs/examples/java/UserManagementService.java
// lines: 188-195
// symbol: UserManagementService#validateEmail
// notes: references java.util.ArrayList, java.util.HashMap, java.util.List, java.util.Map, java.util.Optional, java.util.regex.Pattern; calls EMAIL_PATTERN.matcher, email.trim, isEmpty, matches
    private void validateEmail(String email) {
        if (email == null || email.trim().isEmpty()) {
            throw new IllegalArgumentException("メールアドレスは必須です");
        }
        if (!EMAIL_PATTERN.matcher(email).matches()) {
            throw new IllegalArgumentException("メールアドレスの形式が不正です");
        }
    }

File: code2map/docs/examples/python/output/parts/UserManagementService__validate_email.py
# code2map fragment (non-buildable)
# id: CD16
# original: docs/examples/python/user_management_service.py
# lines: 217-221
# symbol: UserManagementService#_validate_email
# notes: references dataclasses.dataclass, re, typing.Optional; calls ValueError, email.strip, self.EMAIL_PATTERN.match
    def _validate_email(self, email: str) -> None:
        if not email or not email.strip():
            raise ValueError("メールアドレスは必須です")
        if not self.EMAIL_PATTERN.match(email):
            raise ValueError("メールアドレスの形式が不正です")
